src/tutorials/0006-anatomy-of-a-cid/02.md

```diff
@@ -0,0 +1,36 @@
+---
+    title: "Multihash"
+    type: "multiple-choice"
+---
+
+Occassionally, a hashing algorithm may be proven to be insecure, meaning it no longer complies with the characteristics that we defined earlier. This has already happened with `sha1`. With time, other algorithms may prove to be insufficient for content addressing in IPFS. For this reason, and in order to support multiple cryptographic algorithms, **we need to be able to know which algorithm was used to generate the hash** of specific content.
+
+![What is the hashing algorithm used in a hash?](tutorial-assets/T0006L02-what-algo.jpg)
+
+So how can we do this?
+To support multiple hashing algorithms, we use **Multihash**.
+
+## Multihash format
+
+A [**multihash**](https://multiformats.io/multihash/) is a self-describing hash which itself contains metadata that describes both its length and what cryptographic algorithm generated it. IPFS is future-proof because it uses multihash to
+**support multiple hashing algorithms** rather than relying on a specific one.
+
+Multihashes follow the `TLV` pattern (`type-length-value`). Essentially, the "original hash" is prefixed with a `type` and the `length` of the hash.
+
+![Multihash format](tutorial-assets/T0006L02-multihash.jpg)
+
+ - `type`: identifier of the **cryptographic algorithm** used to generate the hash (e.g. the identifier of `sha2-256` would be `18` - `0x12` in hexadecimal) - see the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv) for all the identifiers
+ - `length`: the actual **length** of the hash (using `sha2-256` it would be `256`)
```

Comment:
@olizilla Could you shed some light on why the  CID Inspector expresses length in `bits` vs `bytes`? @ribasushi is suggesting switching to `bytes` but we're explaining the CID Inspector readout here so we'd confuse the reader by showing something different from what's listed there.

Reply:
@terichadbourne note that I am not specifically advocating for switching from bits to bytes in the *visualiser*. It is the on-wire format where everything is unquiestinably in bytes ( this also has an implication that you can not take 255 bits of the hash: you can do either 248 or 256 and that's that )

Reply:
I made the CID inspector show bits as it was heavily inspired by https://github.com/multiformats/multihash#idea-self-describe-the-values-to-distinguish

I don't have a strong opinion on if the length should be show in `bits` or `bytes` as long as it includes the unit. bits seems popular when describing the length of hash functions https://en.wikipedia.org/wiki/List_of_hash_functions

Reply:
The particular lesson in the tutorial is about showing the CID Inspector and explaining what you'll find there, so I'd love to match what's there as opposed to introducing concepts that aren't there. 

To @olizilla's point, it's much easier for me to understand sha-256 matching something expressed as 256 X than as 32 Y.  "Normal people" don't know how to convert bits to bytes in their head so they won't connect 32 and 256 on their own. 

@ribasushi perhaps you could put in a PR to the CID Inspector to add a bytes field _in addition to_ the bits field to appease both the technical purists and the newcomers? And once it's there we could update the tutorial content to mention both?

Reply:
@terichadbourne I think we miscommunicated somewhere. I believe the CID Inspector **is doing the right thing** - i.e. speaking of bits in the context of hashes. In other words: what @olizilla said.

What I was highlighting is that **on the wire** you have to work in bytes. Since a number of your images show the on-wire binary, you need to:
- fix the images to show 0x20
- (optionally) explain why 0x20 and 256 are the same thing in this context

TLDR:
- when visualizing: speak in bits
- when talking about "what is *in* the CID" - you need to show/mention bytes ( otherwise https://github.com/ProtoSchool/protoschool.github.io/pull/359#discussion_r384689415 happens )

Reply:
@ribasushi To clarify, is it that there are pictures of 1s and 0s that have the wrong order of 1s and 0s in them because some conversion was done incorrectly or that you believe certain parts of the  1s and 0s pics should be labelled as "0x20" instead of labelled as "256"?

Reply:
Very good point @ribasushi.
@alanshaw mentioned in the original talk that the representation is like so (1st byte = 128, 2nd byte = 2) because of the varint representation. But I thought it was not necessary to bring that up in this tutorial.

So regarding the length mentions, we should change it to bytes (show the conversion from bits to bytes like you said) - 100% agree.
Regarding the binary representation, do you think it would make sense to put the two bytes together and point to `256` then? Maybe with `varint` under parentheses?

Reply:
We had a short Zoom with @zebateira to clarify the context of above remark. Desired outcome is [mention the appropriate/expected unit in the corresponding context](https://github.com/ProtoSchool/protoschool.github.io/pull/359#issuecomment-592529505).

Thanks for taking the time to understand the subtle mismatch!